src/openrct2/ride/Vehicle.cpp

```diff
@@ -4671,6 +4673,8 @@ static void vehicle_update_boat_location(rct_vehicle* vehicle)
 static bool vehicle_boat_is_location_accessible(const TileCoordsXYZ& location)
 {
     TileElement* tileElement = map_get_first_element_at(location.x, location.y);
+    if (tileElement == nullptr)
+        return true;
```

Comment:
should return false as the location doesn't exist so can't be accessed. Ultimately will never happen but best to be safe.